python/pyarrow/tests/test_dataset.py

```diff
@@ -3621,6 +3621,204 @@ def compare_tables_ignoring_order(t1, t2):
+        batches = table.to_batches()
+        for batch in batches:
+            batched_data.append(batch.num_rows)
+
+    assert batched_data[0] > min_rows_per_group and \
+        batched_data[0] <= max_rows_per_group
+    assert batched_data[1] > min_rows_per_group and \
+        batched_data[1] <= max_rows_per_group
+    assert batched_data[2] <= max_rows_per_group
+
+
+def test_write_dataset_max_rows_per_group(tempdir):
+    directory = tempdir / 'ds'
+    max_rows_per_group = 18
+    num_of_columns = 2
+    records_per_row = 30
+    unique_records = 5
+
+    data, column_names = _generate_data_and_columns(num_of_columns,
+                                                    records_per_row,
+                                                    unique_records)
+
+    record_batch = pa.record_batch(data=data, names=column_names)
+
+    data_source = directory / "max_rows_group"
+
+    ds.write_dataset(data=record_batch, base_dir=data_source,
+                     max_rows_per_group=max_rows_per_group,
+                     format="parquet")
+
+    files_in_dir = os.listdir(data_source)
+    batched_data = []
+    for _, f_file in enumerate(files_in_dir):
+        f_path = data_source / str(f_file)
+        dataset = ds.dataset(f_path, format="parquet")
+        table = dataset.to_table()
+        batches = table.to_batches()
+        for batch in batches:
+            batched_data.append(batch.num_rows)
+
+    assert batched_data == [18, 12]
+
+
+def test_write_dataset_max_open_files(tempdir):
+    # TODO: INCOMPLETE TEST CASE WIP
+    directory = tempdir / 'ds'
+    print("Directory : ", directory)
+
+    record_batch_1 = pa.record_batch(data=[[1, 2, 3, 4, 0],
+                                     ['a', 'b', 'c', 'd', 'e']],
+                                     names=['c1', 'c2'])
+    record_batch_2 = pa.record_batch(data=[[5, 6, 7, 8, 0],
+                                     ['a', 'b', 'c', 'd', 'e']],
+                                     names=['c1', 'c2'])
+    record_batch_3 = pa.record_batch(data=[[9, 10, 11, 12, 0],
+                                     ['a', 'b', 'c', 'd', 'e']],
+                                     names=['c1', 'c2'])
+    record_batch_4 = pa.record_batch(data=[[13, 14, 15, 16, 0],
+                                     ['a', 'b', 'c', 'd', 'e']],
+                                     names=['c1', 'c2'])
+
+    table = pa.Table.from_batches([record_batch_1, record_batch_2,
+                                   record_batch_3, record_batch_4])
+
+    partitioning = ds.partitioning(
+        pa.schema([("c2", pa.string())]), flavor="hive")
+
+    data_source_1 = directory / "default"
+
+    ds.write_dataset(data=table, base_dir=data_source_1,
+                     partitioning=partitioning, format="parquet")
+
+    # CASE 1: when max_open_files=default & max_open_files >= num_of_partitions
+    #         the number of unique rows must be equal to
+    #         the number of files generated
```

Comment:
I think there is a misinterpretation in the comment. It must be 

```python
     # CASE 1: when max_open_files=default & max_open_files >= num_of_partitions
     # In case of a writing to disk via partitioning based on a particular column (considering row labels in that column), 
     # the number of unique rows must be equal to the number of files generated
```

Reply:
I think my confusion what what you meant by "unique rows". You mean unique based on value of `c2`, not `c1` and `c2`, right? So each file will have multiple rows, but there will only be one file per value of `c2` (e.g. 1 file for `a`, one file for `b`, etc.). Do I have that right?

Reply:
Yes, that's right.